patchwork-registries/src/main/java/net/minecraftforge/registries/ForgeRegistry.java

```diff
@@ -204,4 +272,72 @@ public int hashCode() {
 			return identifier.hashCode() * 33 + value.hashCode();
 		}
 	}
+
+	/**
+	 * Used to control the times where people can modify this registry.
+	 * Users should only ever register things in the Register<?> events!
+	 */
+	public void freeze() {
+		this.isFrozen = true;
+	}
+
+	public void unfreeze() {
+		this.isFrozen = false;
+	}
+
+	@Override
+	public boolean isLocked() {
+		return this.isFrozen;
+	}
+
+	@Override
+	public void clear() {
+		if (!this.isModifiable) {
+			throw new UnsupportedOperationException("Attempted to clear a non-modifiable Forge Registry");
+		}
+
+		if (this.isLocked()) {
+			throw new IllegalStateException("Attempted to clear the registry to late.");
+		}
+
+		if (!this.isVanilla) {
+			throw new UnsupportedOperationException("Vanilla registry does not support clear().");
```

Comment:
All registries should be able to support clearing with the help of a mixin (see my other comment)

Reply:
Could clearing vanilla registries break something?